docs/adding_a_task.md

```diff
@@ -0,0 +1,252 @@
+
+# Adding a Task
+
+To add a new task to MTEB, you need to do three things:
+
+1) Implement a task with the desired dataset, by subclassing an abstract task
+2) Add metadata to the task
+3) Submit the edits to the [MTEB](https://github.com/embeddings-benchmark/mteb) repository
+
+If you have any questions regarding this process feel free to open a discussion [thread](https://github.com/embeddings-benchmark/mteb/discussions). 
+
+> Note: When we mention adding a dataset we refer to a subclass of one of the abstasks.
+
+## 1) Creating a new subclass
+
+### A Simple Example
+
+To add a new task, you need to implement a new class that inherits from the `AbsTask` associated with the task type (e.g. `AbsTaskReranking` for reranking tasks). You can find the supported task types in [here](https://github.com/embeddings-benchmark/mteb-draft/tree/main/mteb/abstasks).
+
+```python
+from mteb import MTEB
+from mteb.abstasks.AbsTaskReranking import AbsTaskReranking
+from sentence_transformers import SentenceTransformer
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+class SciDocsReranking(AbsTaskReranking):
+    metadata = TaskMetadata(
+        name="SciDocsRR",
+        description="Ranking of related scientific papers based on their title.",
+        reference="https://allenai.org/data/scidocs",
+        hf_hub_name="mteb/scidocs-reranking",
+        type="Reranking",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["en"],
+        main_score="map",
+        revision="d3c5e1fc0b855ab6097bf1cda04dd73947d7caab",
+        date=None,
+        form="written",
+        domains=["Academic", "Non-fiction"],
+        task_subtypes=["Scientific Reranking"],
+        license="cc-by-4.0",
+        socioeconomic_status="high",
+        annotations_creators=None,
+        dialect=None,
+        text_creation="found",
+        bibtex_citation= ... # removed for brevity
+)
+
+# testing the task with a model:
+model = SentenceTransformer("average_word_embeddings_komninos")
+evaluation = MTEB(tasks=[MindSmallReranking()])
+evaluation.run(model)
+```
+
+> **Note:** for multilingual tasks, make sure your class also inherits from the `MultilingualTask` class like in [this](https://github.com/embeddings-benchmark/mteb-draft/blob/main/mteb/tasks/Classification/MTOPIntentClassification.py) example.  
+> For cross-lingual tasks, make sure your class also inherits from the `CrosslingualTask` class like in [this](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/BitextMining/TatoebaBitextMining.py).
+
+
+
+### A Detailed Example
+Often the dataset from HuggingFace is not in the format expected by MTEB. To resolve this you can add a `dataset_transform` method to your dataset to ensure it is in the right format. Here is an example along with some design considerations: 
```

Comment:
```suggestion
Often the dataset from HuggingFace is not in the format expected by MTEB. To resolve this you can either change the format on Hugging Face or add a `dataset_transform` method to your dataset to transform it into the right format on the fly. Here is an example along with some design considerations: 
```

Reply:
by "change the format on Hugging Face" you mean uploading the dataset again with the expected format ?

Reply:
Either that or just modifying it if it's their HF dataset anyways